sw/device/tests/power_virus_systemtest.c

```diff
@@ -0,0 +1,173 @@
+ *
+ * (Task 3) Activate All Blocks
+ *    1. start OTBN operations
+ *    2. start KMAC operations
+ *    3. start AES operations
+ *    4. start HMAC operations
+ *    5. start I2C operations
+ *    6. start UART operations
+ *    7. set GPIO 0 high (to indicate max power state)
+ *    8. wait until complete
+ *    9. set GPIO 0 lot (to indicate out of max power state)
+ *
+ * Internal Test Parameters:
+ *  - adc_ctrl power-up time
+ *  - internal vs. external clock source
+ *
+ * OTP Test Parameters:
+ *  - turn on icache
+ *
+ * (External) Host Side Test Component:
+ *    1. drive spi_device during time of max power consumption
+ */
+
+OTTF_DEFINE_TEST_CONFIG(.enable_concurrency = true,
+                        .can_clobber_uart = false, );
+
+/**
+ * Peripheral DIF Handles.
+ */
+static dif_pinmux_t pinmux;
+static dif_gpio_t gpio;
+static dif_adc_ctrl_t adc_ctrl;
+
+/**
+ * Test configuration parameters.
+ */
+enum {
+  /**
+   * ADC controller parameters.
+   */
+  kAdcCtrlPowerUpTimeAonCycles = 15,  // maximum power-up time
+};
+
+/**
+ * Initializes all DIF handles for each peripheral used in this test.
+ */
+static void init_peripheral_handles() {
+  // adc_ctrl
+  CHECK_DIF_OK(dif_adc_ctrl_init(
+      mmio_region_from_addr(TOP_EARLGREY_ADC_CTRL_AON_BASE_ADDR), &adc_ctrl));
+  // gpio
+  CHECK_DIF_OK(
+      dif_gpio_init(mmio_region_from_addr(TOP_EARLGREY_GPIO_BASE_ADDR), &gpio));
+  // pinmux
+  CHECK_DIF_OK(dif_pinmux_init(
+      mmio_region_from_addr(TOP_EARLGREY_PINMUX_AON_BASE_ADDR), &pinmux));
+}
+
+/**
+ * Configures GPIO 0 (mapped to pad IOA2) as an indicator pin, to go high during
+ * the power state(s) of interest.
+ */
+static void configure_gpio_indicator_pin() {
+  CHECK_DIF_OK(dif_pinmux_output_select(&pinmux, kTopEarlgreyPinmuxMioOutIoa2,
+                                        kTopEarlgreyPinmuxOutselGpioGpio0));
+  CHECK_DIF_OK(
+      dif_gpio_output_set_enabled(&gpio, /*pin=*/0, kDifToggleEnabled));
+}
+
+/**
+ * Configures adc_ctrl to continuously sample data (applying all filters across
+ * both channels) in normal power mode, which is the most power intensive
+ * sampling mode.
+ */
+static void configure_adc_ctrl_to_continuously_sample() {
+  CHECK_DIF_OK(dif_adc_ctrl_configure(
+      &adc_ctrl,
+      (dif_adc_ctrl_config_t){
+          .mode = kDifAdcCtrlNormalPowerScanMode,
+          .power_up_time_aon_cycles = kAdcCtrlPowerUpTimeAonCycles,
+          // Below configurations are unused, so set them to their reset values.
+          .wake_up_time_aon_cycles = ADC_CTRL_ADC_PD_CTL_WAKEUP_TIME_MASK,
+          .num_low_power_samples = ADC_CTRL_ADC_LP_SAMPLE_CTL_REG_RESVAL,
+          .num_normal_power_samples = ADC_CTRL_ADC_SAMPLE_CTL_REG_RESVAL,
+      }));
+  for (int filter = 0; filter < ADC_CTRL_PARAM_NUM_ADC_FILTER; ++filter) {
+    for (int channel = 0; channel < ADC_CTRL_PARAM_NUM_ADC_CHANNEL; ++channel) {
+      LOG_INFO("Configuring filter %d on channel %d.");
```

Comment:
```suggestion
      LOG_INFO("Configuring filter %d on channel %d.", filter, channel);
```

Reply:
fixed (removed this LOG, it is unnecessary)